lib/reports/subscriber_lists_report.rb

```diff
@@ -12,32 +12,50 @@ class Reports::SubscriberListsReport
                    matched_content_changes_for_date
                    matched_messages_for_date].freeze
 
-  def initialize(date)
+  def initialize(date, slugs: "", tags_pattern: nil, links_pattern: nil)
     @date = Time.zone.parse(date)
+    @slugs = slugs.split(",")
+    @tags_pattern = tags_pattern
+    @links_pattern = links_pattern
   end
 
   def call
     validate_date
+    validate_slugs
 
-    CSV do |csv|
+    CSV.generate do |csv|
       csv << CSV_HEADERS
-
-      SubscriberList.where("created_at < ?", date.end_of_day).find_each do |list|
-        csv << export_list_row(list)
-      end
+      lists_to_report.find_each { |list| csv << export_list_row(list) }
     end
   end
 
 private
 
+  def lists_to_report
+    scope = SubscriberList.where("created_at < ?", date.end_of_day)
+    scope = scope.where(slug: slugs) if slugs.any?
+    scope = scope.where("tags::text like ?", "%#{tags_pattern}%") if tags_pattern
+    scope = scope.where("links::text like ?", "%#{links_pattern}%") if links_pattern
+    scope
+  end
+
   def validate_date
     raise "Invalid date" if date.blank?
     raise "Date must be in the past" if date >= Time.zone.today
     raise "Date must be within a year old" if date <= 1.year.ago
   end
 
+  def validate_slugs
+    return unless slugs
```

Comment:
This looks like it missed a tweak in the conversion to string. Should it be `if slugs.empty?`?

Reply:
Oh yeah. I realise we don't need that line at all anymore - removed!